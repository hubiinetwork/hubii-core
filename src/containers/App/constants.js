export const DEFAULT_LOCALE = 'en';

export const NOTIFY = 'src/App/NOTIFY';
export const SHOW_RELEASE_NOTES = 'src/App/SHOW_RELEASE_NOTES';
export const HIDE_RELEASE_NOTES = 'src/App/HIDE_RELEASE_NOTES';
export const LOAD_RELEASE_NOTES = 'src/App/LOAD_RELEASE_NOTES';
export const LOAD_RELEASE_NOTES_SUCCESS = 'src/App/LOAD_RELEASE_NOTES_SUCCESS';
export const INSTALL_NEW_RELEASE = 'src/App/INSTALL_NEW_RELEASE';
