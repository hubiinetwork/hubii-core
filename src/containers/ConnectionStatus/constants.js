export const NETWORK_FAILURE = 'src/containers/ConnectionStatus/NETWORK_FAILURE';
export const NETWORK_RECONNECTED = 'src/containers/ConnectionStatus/NETWORK_RECONNECTED';
