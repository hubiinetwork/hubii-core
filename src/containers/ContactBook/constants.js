/*
 *
 * WalletManager constants
 *
 */

export const LOAD_CONTACTS = 'src/contacts/LOAD_CONTACTS';
export const LOAD_CONTACTS_SUCCESS = 'src/contacts/LOAD_CONTACTS_SUCCESS';

export const CREATE_CONTACT = 'src/contacts/CREATE_CONTACT';
export const CREATE_CONTACT_SUCCESS = 'src/contacts/CREATE_CONTACT_SUCCESS';

export const REMOVE_CONTACT = 'src/contacts/REMOVE_CONTACT';
export const REMOVE_CONTACT_SUCCESS = 'src/contacts/REMOVE_CONTACT_SUCCESS';

export const EDIT_CONTACT = 'src/contacts/EDIT_CONTACT';
export const EDIT_CONTACT_SUCCESS = 'src/contacts/EDIT_CONTACT_SUCCESS';
