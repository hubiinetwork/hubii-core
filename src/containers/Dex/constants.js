
export const LOAD_PRICE_HISTORY = 'src/dex/LOAD_PRICE_HISTORY';
export const LOAD_PRICE_HISTORY_SUCCESS = 'src/dex/LOAD_PRICE_HISTORY_SUCCESS';
export const LOAD_PRICE_HISTORY_ERROR = 'src/dex/LOAD_PRICE_HISTORY_ERROR';
export const LISTEN_LATEST_PRICE = 'src/dex/LISTEN_LATEST_PRICE';
export const UPDATE_LATEST_PRICE_SUCCESS = 'src/dex/UPDATE_LATEST_PRICE_SUCCESS';
export const UPDATE_LATEST_PRICE_ERROR = 'src/dex/UPDATE_LATEST_PRICE_ERROR';
