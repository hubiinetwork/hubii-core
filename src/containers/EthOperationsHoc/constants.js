/*
 *
 * EthOperationsHoc constants
 *
 */


export const LOAD_BLOCK_HEIGHT = 'src/EthOperationsHoc/LOAD_BLOCK_HEIGHT';
export const LOAD_BLOCK_HEIGHT_SUCCESS = 'src/EthOperationsHoc/LOAD_BLOCK_HEIGHT_SUCCESS';
export const LOAD_BLOCK_HEIGHT_ERROR = 'src/EthOperationsHoc/LOAD_BLOCK_HEIGHT_ERROR';
export const LOAD_GAS_STATISTICS = 'src/EthOperationsHoc/LOAD_GAS_STATISTICS';
export const LOAD_GAS_STATISTICS_SUCCESS = 'src/EthOperationsHoc/LOAD_GAS_STATISTICS_SUCCESS';
export const LOAD_GAS_STATISTICS_ERROR = 'src/EthOperationsHoc/LOAD_GAS_STATISTICS_ERROR';
