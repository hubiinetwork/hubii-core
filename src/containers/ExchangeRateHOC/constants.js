export const LOAD_EXCHANGE_RATE = 'app/ExchangeRate/LOAD_EXCHANGE_RATE';
export const LOAD_EXCHANGE_RATE_SUCCESS = 'app/ExchangeRate/LOAD_EXCHANGE_RATE_SUCCESS';
export const LOAD_EXCHANGE_RATE_FAILURE = 'app/ExchangeRate/LOAD_EXCHANGE_RATE_FAILURE';
export const LOAD_TICKERS_SUCCESS = 'app/ExchangeRate/LOAD_TICKERS_SUCCESS';
export const LOAD_TICKERS_FAILURE = 'app/ExchangeRate/LOAD_TICKERS_FAILURE';
