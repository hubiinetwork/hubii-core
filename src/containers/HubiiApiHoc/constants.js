/*
 *
 * HubiiApiHoc constants
 *
 */

export const INIT_HUBII_API = 'src/HubiiApiHoc/INIT_HUBII_API';

export const LOAD_WALLET_BALANCES = 'src/HubiiApiHoc/LOAD_WALLET_BALANCES';
export const LOAD_WALLET_BALANCES_SUCCESS = 'src/HubiiApiHoc/LOAD_WALLET_BALANCES_SUCCESS';
export const LOAD_WALLET_BALANCES_ERROR = 'src/HubiiApiHoc/LOAD_WALLET_BALANCES_ERROR';

export const LOAD_SUPPORTED_TOKENS = 'src/HubiiApiHoc/LOAD_SUPPORTED_TOKENS';
export const LOAD_SUPPORTED_TOKENS_SUCCESS = 'src/HubiiApiHoc/LOAD_SUPPORTED_TOKENS_SUCCESS';
export const LOAD_SUPPORTED_TOKENS_ERROR = 'src/HubiiApiHoc/LOAD_SUPPORTED_TOKENS_ERROR';

export const LOAD_PRICES = 'src/HubiiApiHoc/LOAD_PRICES';
export const LOAD_PRICES_SUCCESS = 'src/HubiiApiHoc/LOAD_PRICES_SUCCESS';
export const LOAD_PRICES_ERROR = 'src/HubiiApiHoc/LOAD_PRICES_ERROR';

export const LOAD_TRANSACTIONS = 'src/HubiiApiHoc/LOAD_TRANSACTIONS';
export const LOAD_TRANSACTIONS_SUCCESS = 'src/HubiiApiHoc/LOAD_TRANSACTIONS_SUCCESS';
export const LOAD_TRANSACTIONS_ERROR = 'src/HubiiApiHoc/LOAD_TRANSACTIONS_ERROR';
