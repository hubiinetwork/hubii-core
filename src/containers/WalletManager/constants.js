/*
 *
 * WalletManager constants
 *
 */

export const CREATE_NEW_WALLET = 'src/WalletManager/CREATE_NEW_WALLET';
export const CREATE_NEW_WALLET_SUCCESS = 'src/WalletManager/CREATE_NEW_WALLET_SUCCESS';
export const CREATE_NEW_WALLET_FAILURE = 'src/WalletManager/CREATE_NEW_WALLET_FAILURE';

export const UPDATE_PROGRESS = 'src/WalletManager/UPDATE_PROGRESS';

export const DECRYPT_WALLET = 'src/WalletManager/DECRYPT_WALLET';
export const DECRYPT_WALLET_SUCCESS = 'src/WalletManager/DECRYPT_WALLET_SUCCESS';
export const DECRYPT_WALLET_FAILURE = 'src/WalletManager/DECRYPT_WALLET_FAILURE';
