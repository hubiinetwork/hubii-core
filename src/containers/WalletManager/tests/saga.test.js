/**
 * WalletManager sagas
 */

import { Wallet } from 'ethers';

/* eslint-disable redux-saga/yield-effects */
import { takeEvery, put } from 'redux-saga/effects';
import walletManager, { createWallet, decryptWallet, progressCallback } from '../saga';
import { CREATE_NEW_WALLET, DECRYPT_WALLET } from '../constants';
import {
  createNewWalletSuccess,
  createNewWalletFailed,
  decryptWalletSuccess,
  decryptWalletFailed,
  updateProgress,
} from '../actions';

describe('createWallet saga', () => {
  let mnemonic = 'movie viable write punch mango arrest cotton page grass dad document practice';
  let derivationPath = 'm/44\'/60\'/0\'/0/0';
  const decryptedWallet = Wallet.fromMnemonic(mnemonic, derivationPath);
  const name = 'wallet8';
  let encryptedWallet;
  let password;
  beforeEach(() => {
    password = 'dogs';
    mnemonic = 'movie viable write punch mango arrest cotton page grass dad document practice';
    derivationPath = 'm/44\'/60\'/0\'/0/0';
    encryptedWallet = '{"address":"a0eccd7605bb117dd2a4cd55979c720cf00f7fa4","id":"f17128a6-c5f0-4af0-a168-67cf6d3d8552","version":3,"Crypto":{"cipher":"aes-128-ctr","cipherparams":{"iv":"6167c13fe3cd195b4ce9312a9f9399ce"},"ciphertext":"2434b52afa29851edea2acb7f33dd854fc7e7b036ad6a2c3614f3d61ef0e19ce","kdf":"scrypt","kdfparams":{"salt":"b0662c8968389207137be9f346fb1cfba604f9d214e95012881025b7ebc5b9da","n":131072,"dklen":32,"p":1,"r":8},"mac":"256bd09baf3341e9f7df675b8a8cc551b86dfd0dfdf1aa8df2596882f3751496"},"x-ethers":{"client":"ethers.js","gethFilename":"UTC--2018-06-19T04-19-27.0Z--a0eccd7605bb117dd2a4cd55979c720cf00f7fa4","mnemonicCounter":"20da552ff9e584fc89194af19543a096","mnemonicCiphertext":"ff46b728607532d5be86a0647b169a18","version":"0.1"}}';
  });

  it('should dispatch the createNewWalletSuccess action if successful', () => {
    const createWalletGenerator = createWallet({ name, mnemonic, derivationPath, password });
    const callDescriptor = createWalletGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
    const putDescriptor = createWalletGenerator.next(encryptedWallet).value;
    expect(JSON.stringify(putDescriptor)).toEqual(JSON.stringify(put(createNewWalletSuccess(name, encryptedWallet, decryptedWallet))));
  });

  it('should dispatch the createNewWalletFailed action if bad mnemonic', () => {
    mnemonic = 'rubbish';
    const createWalletGenerator = createWallet({ name, mnemonic, derivationPath, password });
    const putDescriptor = createWalletGenerator.next().value;
    const error = new Error('invalid mnemonic');
    expect(putDescriptor).toEqual(put(createNewWalletFailed(error)));
  });

  it('should dispatch the createNewWalletFailed action if bad derivationPath', () => {
    derivationPath = 'm/0.0';
    const createWalletGenerator = createWallet({ name, mnemonic, derivationPath, password });
    const putDescriptor = createWalletGenerator.next().value;
    const error = new Error('invlaid path component - 0.0');
    expect(putDescriptor).toEqual(put(createNewWalletFailed(error)));
  });

  it('should dispatch the createNewWalletFailed action if invalid mnemonic password', () => {
    mnemonic = 'rubbish';
    const createWalletGenerator = createWallet({ name, mnemonic, derivationPath, password });
    const putDescriptor = createWalletGenerator.next().value;
    const error = new Error('invalid mnemonic');
    expect(putDescriptor).toEqual(put(createNewWalletFailed(error)));
  });

  it('should dispatch createNewWalletFailed action if name is undefined', () => {
    const createWalletGenerator = createWallet({ mnemonic, derivationPath, password });
    const putDescriptor = createWalletGenerator.next().value;
    const error = new Error('invalid param');
    expect(putDescriptor).toEqual(put(createNewWalletFailed(error)));
  });

  it('should dispatch createNewWalletFailed action if mnemonic is undefined', () => {
    const createWalletGenerator = createWallet({ name, derivationPath, password });
    const putDescriptor = createWalletGenerator.next().value;
    const error = new Error('invalid param');
    expect(putDescriptor).toEqual(put(createNewWalletFailed(error)));
  });

  it('should dispatch createNewWalletFailed action if derivationPath is undefined', () => {
    const createWalletGenerator = createWallet({ mnemonic, name, password });
    const putDescriptor = createWalletGenerator.next().value;
    const error = new Error('invalid param');
    expect(putDescriptor).toEqual(put(createNewWalletFailed(error)));
  });

  it('should dispatch createNewWalletFailed action if password is undefined', () => {
    const createWalletGenerator = createWallet({ mnemonic, name, derivationPath });
    const putDescriptor = createWalletGenerator.next().value;
    const error = new Error('invalid param');
    expect(putDescriptor).toEqual(put(createNewWalletFailed(error)));
  });
});

describe('decryptWallet saga', () => {
  const encryptedWallet = '{"address":"a0eccd7605bb117dd2a4cd55979c720cf00f7fa4","id":"f17128a6-c5f0-4af0-a168-67cf6d3d8552","version":3,"Crypto":{"cipher":"aes-128-ctr","cipherparams":{"iv":"6167c13fe3cd195b4ce9312a9f9399ce"},"ciphertext":"2434b52afa29851edea2acb7f33dd854fc7e7b036ad6a2c3614f3d61ef0e19ce","kdf":"scrypt","kdfparams":{"salt":"b0662c8968389207137be9f346fb1cfba604f9d214e95012881025b7ebc5b9da","n":131072,"dklen":32,"p":1,"r":8},"mac":"256bd09baf3341e9f7df675b8a8cc551b86dfd0dfdf1aa8df2596882f3751496"},"x-ethers":{"client":"ethers.js","gethFilename":"UTC--2018-06-19T04-19-27.0Z--a0eccd7605bb117dd2a4cd55979c720cf00f7fa4","mnemonicCounter":"20da552ff9e584fc89194af19543a096","mnemonicCiphertext":"ff46b728607532d5be86a0647b169a18","version":"0.1"}}';
  const password = 'dogs';
  const decryptedWallet = { privateKey: '0x409300caf64bdf96a92d7f99547a5d67702fbdd759bbea4ca19b11a21d9c8528', defaultGasLimit: 1500000, address: '0xA0EcCD7605Bb117DD2A4Cd55979C720Cf00F7fa4', mnemonic: 'movie viable write punch mango arrest cotton page grass dad document practice', path: "m/44'/60'/0'/0/0" };

  it('should dispatch the decryptWalletSuccess action if successful', () => {
    const decryptWalletGenerator = decryptWallet({ name, encryptedWallet, password });
    const callDescriptor = decryptWalletGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
    const putDescriptor = decryptWalletGenerator.next(decryptedWallet).value;
    expect(JSON.stringify(putDescriptor)).toEqual(JSON.stringify(put(decryptWalletSuccess(name, decryptedWallet))));
  });

  it('should dispatch decryptWalletFailed action if name is undefined', () => {
    const decryptWalletGenerator = decryptWallet({ encryptedWallet, password });
    const putDescriptor = decryptWalletGenerator.next().value;
    const error = new Error('name undefined');
    expect(putDescriptor).toEqual(put(decryptWalletFailed(error)));
  });

  it('should dispatch the decryptWalletFailed action if decryption fails', () => {
    const decryptWalletGenerator = decryptWallet({ name, encryptedWallet, password });
    const callDescriptor = decryptWalletGenerator.next().value;
    const error = new Error('some error occured');
    expect(callDescriptor).toMatchSnapshot();
    const putDescriptor = decryptWalletGenerator.next(error).value;
    expect(putDescriptor).toEqual(put(decryptWalletFailed(error)));
  });
});

describe('progressCallback', () => {
  it('should dispatch the updateProgress action on valid input', () => {
    const progress = 0.3423;
    const progressCallbackGenerator = progressCallback(progress);
    const putDescriptor = progressCallbackGenerator.next().value;
    expect(putDescriptor).toEqual(put(updateProgress(progress)));
  });

  it('should throw error on negative input', () => {
    const progress = -0.3423;
    expect(() => {
      const progressCallbackGenerator = progressCallback(progress);
      progressCallbackGenerator.next();
    }).toThrow();
  });

  it('should throw error on input > 1', () => {
    const progress = 1.3;
    expect(() => {
      const progressCallbackGenerator = progressCallback(progress);
      progressCallbackGenerator.next();
    }).toThrow();
  });
});

describe('root Saga', () => {
  const walletManagerSaga = walletManager();

  it('should start task to watch for CREATE_NEW_WALLET action', () => {
    const takeDescriptor = walletManagerSaga.next().value;
    expect(takeDescriptor).toEqual(takeEvery(CREATE_NEW_WALLET, createWallet));
  });

  it('should start task to watch for DECRYPT_WALLET action', () => {
    const takeDescriptor = walletManagerSaga.next().value;
    expect(takeDescriptor).toEqual(takeEvery(DECRYPT_WALLET, decryptWallet));
  });
});
