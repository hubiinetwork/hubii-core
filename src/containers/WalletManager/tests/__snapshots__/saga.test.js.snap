// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createWallet saga should dispatch the createNewWalletSuccess action if successful 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "dogs",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`decryptWallet saga should dispatch the decryptWalletFailed action if decryption fails 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "{\\"address\\":\\"a0eccd7605bb117dd2a4cd55979c720cf00f7fa4\\",\\"id\\":\\"f17128a6-c5f0-4af0-a168-67cf6d3d8552\\",\\"version\\":3,\\"Crypto\\":{\\"cipher\\":\\"aes-128-ctr\\",\\"cipherparams\\":{\\"iv\\":\\"6167c13fe3cd195b4ce9312a9f9399ce\\"},\\"ciphertext\\":\\"2434b52afa29851edea2acb7f33dd854fc7e7b036ad6a2c3614f3d61ef0e19ce\\",\\"kdf\\":\\"scrypt\\",\\"kdfparams\\":{\\"salt\\":\\"b0662c8968389207137be9f346fb1cfba604f9d214e95012881025b7ebc5b9da\\",\\"n\\":131072,\\"dklen\\":32,\\"p\\":1,\\"r\\":8},\\"mac\\":\\"256bd09baf3341e9f7df675b8a8cc551b86dfd0dfdf1aa8df2596882f3751496\\"},\\"x-ethers\\":{\\"client\\":\\"ethers.js\\",\\"gethFilename\\":\\"UTC--2018-06-19T04-19-27.0Z--a0eccd7605bb117dd2a4cd55979c720cf00f7fa4\\",\\"mnemonicCounter\\":\\"20da552ff9e584fc89194af19543a096\\",\\"mnemonicCiphertext\\":\\"ff46b728607532d5be86a0647b169a18\\",\\"version\\":\\"0.1\\"}}",
      "dogs",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`decryptWallet saga should dispatch the decryptWalletSuccess action if successful 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "{\\"address\\":\\"a0eccd7605bb117dd2a4cd55979c720cf00f7fa4\\",\\"id\\":\\"f17128a6-c5f0-4af0-a168-67cf6d3d8552\\",\\"version\\":3,\\"Crypto\\":{\\"cipher\\":\\"aes-128-ctr\\",\\"cipherparams\\":{\\"iv\\":\\"6167c13fe3cd195b4ce9312a9f9399ce\\"},\\"ciphertext\\":\\"2434b52afa29851edea2acb7f33dd854fc7e7b036ad6a2c3614f3d61ef0e19ce\\",\\"kdf\\":\\"scrypt\\",\\"kdfparams\\":{\\"salt\\":\\"b0662c8968389207137be9f346fb1cfba604f9d214e95012881025b7ebc5b9da\\",\\"n\\":131072,\\"dklen\\":32,\\"p\\":1,\\"r\\":8},\\"mac\\":\\"256bd09baf3341e9f7df675b8a8cc551b86dfd0dfdf1aa8df2596882f3751496\\"},\\"x-ethers\\":{\\"client\\":\\"ethers.js\\",\\"gethFilename\\":\\"UTC--2018-06-19T04-19-27.0Z--a0eccd7605bb117dd2a4cd55979c720cf00f7fa4\\",\\"mnemonicCounter\\":\\"20da552ff9e584fc89194af19543a096\\",\\"mnemonicCiphertext\\":\\"ff46b728607532d5be86a0647b169a18\\",\\"version\\":\\"0.1\\"}}",
      "dogs",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
